### PR TITLE
List of initial conditions

### PR DESCRIPTION
**Pull request by gassmoeller**
Very preliminary and not working. Do not bother to review. @jaustermann feel free to take a look and continue if you feel like it. I will talk to you before continuing working on it (at the earliest tomorrow).

### CODE REVIEW

**Comment by jaustermann** [comment]
Just to document what Rene and I just talked about regarding this pull request and issue #927: Doing this properly would be to have a set of initial conditions that are only depth dependent profiles (e.g. adiabatic, adiabatic with boundary layers, constant, function, profile - geotherm - from text file) and a set of initial conditions that are perturbations (e.g. harmonic, S40RTS, SAVANI, function, perturbation from text file). The user could then specify combinations of these, which would normally be one depth dependent profile and one perturbation. The issue with this approach is that it breaks backward compatibility. Due to that we decided to wait with this change until the 2.0 release. In the mean time, the basemodel approach as implemented by @anne-glerum will allow to combine models, which is what I plan to implement for S40RTS and SAVANI.
A similar issue and approach to solving it applies to the compositional fields.

**Comment by gassmoeller** [comment]
This tries to address #927 but needs more work.

**Comment by gassmoeller** [comment]
I think this is finally ready for a review. Addresses part of #1090.

File: include/aspect/initial_conditions/interface.h [hunk 1]
```diff
@@ -93,63 +96,139 @@ namespace aspect
...
+        if (InitialTemperatureType *x = dynamic_cast<InitialTemperatureType *> ( (*p).get()) )
+          return x;
+      return NULL;
+    }
+
 
 
     /**
-     * Return a string that consists of the names of initial conditions that can
+     * Return a string that consists of the names of initial temperature models that can
```
Comment (bangerth) [inline include/aspect/initial_conditions/interface.h]: good catch!

File: source/initial_conditions/interface.cc [hunk 2]
```diff
@@ -85,68 +91,127 @@ namespace aspect
...
+          = Utilities::split_string_list(prm.get("List of model names"));
+
+        AssertThrow(Utilities::has_unique_entries(model_names),
+                    ExcMessage("The list of strings for the parameter "
+                               "'Initial conditions/List of model names' contains entries more than once. "
+                               "This is not allowed. Please check your parameter file."));
+
+        const std::string model_name = prm.get ("Model name");
+
+        AssertThrow (model_name == "unspecified" || model_names.size() == 0,
```
Comment (bangerth) [inline source/initial_conditions/interface.cc]: Nice way to make it backward compatible. I think we should open github issues for all places where we have this kind of compatibility code to remind ourselves that we want to remove them again post-2.0. Can you do that?

File: source/initial_conditions/interface.cc [hunk 3]
```diff
@@ -85,68 +91,127 @@ namespace aspect
...
+                          "will be used to initialize the temperature. "
+                          "The results of each of these models will be added.\n\n"
+                          "The following initial tempeature models are available:\n\n"
+                          +
+                          std_cxx11::get<dim>(registered_plugins).get_description_string());
+
         prm.declare_entry ("Model name", "unspecified",
                            Patterns::Selection (pattern_of_names+"|unspecified"),
                            "Select one of the following models:\n\n"
+                           "Warning: This is the old formulation of specifying "
```
Comment (bangerth) [inline source/initial_conditions/interface.cc]: I think you want to move these three new lines below the list of descriptions -- the sentence structure doesn't work right now.

This is latex, so you can do `\\textbf{Warning:}`. Also, it's not entirely clear what "This" refers to here -- say "Warning: This parameter provides an old and deprecated way for specifying initial temperature models. Use 'List of model names' instead."

File: source/simulator/simulator_access.cc [hunk 4]
```diff
@@ -496,9 +496,17 @@ namespace aspect
   const InitialConditions::Interface<dim> &
   SimulatorAccess<dim>::get_initial_conditions () const
   {
-    Assert (simulator->initial_conditions.get() != 0,
-            ExcMessage("You can not call this function if no such model is actually available."));
-    return *simulator->initial_conditions.get();
+    Assert (get_initial_temperature_manager().get_active_initial_temperature_conditions().size() == 1,
+            ExcMessage("You can only this function if only one initial temperature plugin is active."));
```
Comment (bangerth) [inline source/simulator/simulator_access.cc]: -> you can only CALL this function...

**Pull request merged**